pvlib/albedo.py

```diff
@@ -0,0 +1,103 @@
+
+    Inputs to the model are the water color coefficient :math:`c` [-], the
+    water wave roughness coefficient :math:`r` [-] and the solar elevation
+    :math:`\alpha` [degrees]. Parameters are provided in [1]_ , and are coded
+    for convenience in :data:`~pvlib.albedo.WATER_COLOR_COEFFS` and
+    :data:`~pvlib.albedo.WATER_ROUGHNESS_COEFFS`. The values of these
+    coefficients are experimentally determined.
+
+    +---------------------------------------------------------------+-------------------------------+----------------------------------------+
+    | Surface and condition                                         | Color coefficient (:math:`c`) | Wave roughness coefficient (:math:`r`) |
+    +===============================================================+===============================+========================================+
+    | Clear water no waves                                          | 0.13                          | 0.29                                   |
+    +---------------------------------------------------------------+-------------------------------+----------------------------------------+
+    | Clear water ripples up to 2.5 cm                              | 0.16                          | 0.70                                   |
+    +---------------------------------------------------------------+-------------------------------+----------------------------------------+
+    | Clear water ripples larger than 2.5 cm (occasional whitecaps) | 0.23                          | 1.23                                   |
+    +---------------------------------------------------------------+-------------------------------+----------------------------------------+
+    | Clear water frequent whitecaps                                | 0.30                          | 2.00                                   |
+    +---------------------------------------------------------------+-------------------------------+----------------------------------------+
+    | Green water ripples up to 2.5cm                               | 0.22                          | 0.70                                   |
+    +---------------------------------------------------------------+-------------------------------+----------------------------------------+
+    | Muddy water no waves                                          | 0.19                          | 0.29                                   |
+    +---------------------------------------------------------------+-------------------------------+----------------------------------------+
+
+    References
+    ----------
+    .. [1] Dvoracek M.J., Hannabas B. (1990). "Prediction of albedo for use in
+       evapotranspiration and irrigation scheduling." IN: Visions of the Future
+       American Society of Agricultural Engineers 04-90: 692-699.
+    """  # noqa: E501
+
+    if surface_condition is not None:
+        color_coeff = WATER_COLOR_COEFFS[surface_condition]
```

Comment:
Can you implement any error handling and custom error message here? In case an invalid key is passed.

Reply:
If an incorrect key is specified, the dictionary will raise a KeyError. I think that should be helpful enough and the user can consult the list of `surface_conditions` in the documentation. So I vote not adding any custom error message for invalid keys.